lib/src/simple_tokenizer.dart

```
@@ -0,0 +1,200 @@
+      }
+      if (matchesGroup(match, 10)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 11)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 12)) {
+        return new NgSimpleToken.doubleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 14)) {
+        return new NgSimpleToken.singleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 16)) {
+        return new NgSimpleToken.doubleQuote(offset);
+      }
+      if (matchesGroup(match, 17)) {
+        return new NgSimpleToken.singleQuote(offset);
+      }
+      if (matchesGroup(match, 18)) {
+        return new NgSimpleToken.tagStart(offset);
+      }
+      if (matchesGroup(match, 19)) {
+        return new NgSimpleToken.equalSign(offset);
+      }
+      if (matchesGroup(match, 20)) {
+        return new NgSimpleToken.star(offset);
+      }
+    }
+    return new NgSimpleToken.unexpectedChar(
+        offset, new String.fromCharCode(_scanner.readChar()));
+  }
+
+  NgSimpleToken scanText() {
+    int offset = _scanner.position;
+    if (_scanner.peekChar() == null || _scanner.rest.length == 0) {
+      return new NgSimpleToken.EOF(offset);
+    }
+    if (_scanner.scan(_allTextMatches)) {
+      Match match = _scanner.lastMatch;
+      if (matchesGroup(match, 1)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 2)) {
+        _state = _NgSimpleScannerState.comment;
+        return new NgSimpleToken.commentBegin(offset);
+      }
+      if (matchesGroup(match, 3)) {
+        _state = _NgSimpleScannerState.element;
+        return new NgSimpleToken.tagStart(offset);
+      }
+    }
+    return new NgSimpleToken.unexpectedChar(
```

Might be better to report the error internally and move past it rather than return it

---

I was thinking of keeping the simple tokenizer completely error-agnostic and let the parser that utilizes this tokenizer decide and call all errors. This way we can keep the actual tokenizer as a static const entity.

---

Could be split up into two iterators. The resulting iterators don't have to be const